engine/crates/grafbase-engine/src/resolver_utils/list.rs

```diff
@@ -226,3 +298,122 @@ pub async fn resolve_list_native<'a, T: LegacyOutputType + 'a>(
         Ok(node_id)
     }
 }
+
+/// An iterator over the nullability of lists in a type string
+struct ListNullabilityIter<'a>(Peekable<WrappingTypeIter<'a>>);
+
+impl<'a> ListNullabilityIter<'a> {
+    pub fn new(ty: &'a MetaFieldType) -> Self {
+        ListNullabilityIter(ty.wrapping_types().peekable())
+    }
+}
+
+/// The nullability of a list _and_ its contents
+#[derive(Debug, PartialEq, Clone, Copy)]
+enum ListKind {
+    NullableList(ListInner),
+    NonNullList(ListInner),
+}
+
+impl ListKind {
+    pub fn inner_nullablity(self) -> ListInner {
```

Comment:
Not sure, but seems like having a boolean struct field `has_nullable_item` with a method `has_nullable_item()` would make it simpler than having to deal with an enum.